### PR TITLE
User can create a recipe book

### PR DESCRIPTION
**Pull request by alghazu**
### What we did:
* add `collection` attribute to `User` model
* add `collections_controller`
* add "My Collection" view
* Update authentication for these actions
* add `Add to Collection` button on recipes, only shows to authorized users
* Semaphore: [Build Status](https://semaphoreci.com/craftacademy/food_hub)
* Coveralls: [Coverage Status](https://coveralls.io/github/CraftAcademy/food_hub)

### CODE REVIEW

File: features/user_edits_recipe.1.feature [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  Given I am logged in as "maran@test.com"
+  And I visit the site
+  And I click "Meatball-Stew"
+  When I click "Edit Recipe"
+  And I fill in "Title" with " "
+  And I fill in "Description" with " "
+  And I fill in "Ingredients" with " "
+  And I fill in "Directions" with " "
+  And I click "Update Recipe"
+  Then I should see "Error updating recipe!"
```
Comment (tochman) [inline features/user_edits_recipe.1.feature]: Why is this feature added? Feels out of scope.

File: db/migrate/20180614143214_add_collection_to_user.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddCollectionToUser < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :collection, :string, array: true, default: "{}"
```
Comment (tochman) [inline db/migrate/20180614143214_add_collection_to_user.rb]: I would suggest creating a new class called `Collection` and add a reference to the User class. Should a user have one Collection or many? .

File: features/user_edits_recipe.1.feature (hunk 1 above)
Comment (alghazu) [reply features/user_edits_recipe.1.feature]: It is not supposed to be here, someone must have copy-pasted this by mistake

File: app/controllers/collections_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    current_user.save
+    flash[:notice] = "Recipe was added to collection"
+    redirect_to request.referrer
+  end
+
+  def destroy
+    current_user.collection.recipes.delete(current_recipe)
+    current_user.save
+    flash[:notice] = "Recipe was removed from collection"
+    redirect_to request.referrer
```
Comment (tochman) [inline app/controllers/collections_controller.rb]: Please look into:

```
redirect_back(fallback_location: root_path)
```

Same thing on LOC 12

Comment (tochman) [reply app/controllers/collections_controller.rb]: Also, I think I would prefer to see a notice to the user. Both on `destroy` and `create`

File: app/models/user.rb [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
   has_many :comments
   enum role: [:user, :admin]
-  
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable,
          :omniauthable, omniauth_providers: [:facebook]
   has_many :recipes
+  has_one :collection
+  after_create :make_collection
```
Comment (tochman) [inline app/models/user.rb:10]: What are the arguments that you create a collection this way? When the user is created I mean?

File: app/controllers/collections_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+class CollectionsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @recipes = current_user.collection.recipes
```
Comment (tochman) [inline app/controllers/collections_controller.rb]: This was a bit confusing for me. Both here and in the view (below) I would suggest 

```
@collection = current_user.collection
```
...and then iterate through `@collection.recipes` in the view (or a helper...)

File: app/controllers/collections_controller.rb (hunk 3 above)
Comment (alghazu) [reply app/controllers/collections_controller.rb]: There is a notice to the user on `destroy` and `create`

File: app/models/user.rb (hunk 4 above)
Comment (alghazu) [reply app/models/user.rb:10]: By default every user should have an empty collection, and therefore an instance of `collection` gets created for every new user.
Otherwise, we can create this `collection` instance for the user when he adds a recipe to his collection the first time.

File: app/controllers/collections_controller.rb (hunk 5 above)
Comment (alghazu) [reply app/controllers/collections_controller.rb]: :+1:

File: features/user_can_create_a_recipe_book.feature [hunk 6]
```diff
@@ -0,0 +1,38 @@
+Feature: User can create a recipe book
+  As a user
+  in order to have my favourite recipes easily available
```
Comment (ghost) [inline features/user_can_create_a_recipe_book.feature]: Change ```in``` order to ```In``` order, capital I

**Review by ghost — CHANGES_REQUESTED** [review]
Left one comment in feature file

File: features/user_can_create_a_recipe_book.feature (hunk 6 above)
Comment (alghazu) [reply features/user_can_create_a_recipe_book.feature]: :+1:

**Review by tochman — APPROVED** [review]
Good job!

**Review by ghost — APPROVED** [review]
